traffic_ops/traffic_ops_golang/profile/profiles.go

```diff
@@ -360,3 +362,294 @@ type) VALUES (
+			api.HandleErr(w, r, tx, http.StatusInternalServerError, fmt.Errorf("error: %w in creating profile:%s", err, profile.Name), nil)
+			return
+		}
+		usrErr, sysErr, code := api.ParseDBError(err)
+		api.HandleErr(w, r, tx, code, usrErr, sysErr)
+		return
+	}
+
+	alerts := tc.CreateAlerts(tc.SuccessLevel, "profile was created.")
+	w.Header().Set("Location", fmt.Sprintf("/api/%d.%d/profiles?id=%d", inf.Version.Major, inf.Version.Minor, profile.ID))
+	api.WriteAlertsObj(w, r, http.StatusCreated, alerts, profile)
+	return
+}
+
+// Update a profile for APIv5
+func Update(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, []string{"id"}, []string{"id"})
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	tx := inf.Tx.Tx
+	profile, readValErr := readAndValidateJsonStruct(r)
+	if readValErr != nil {
+		api.HandleErr(w, r, tx, http.StatusBadRequest, readValErr, nil)
+		return
+	}
+
+	//check if user can modify.
+	if len(strings.TrimSpace(profile.CDNName)) != 0 || profile.CDNID != 0 {
+		userErr, sysErr, statusCode := canProfileBeAlteredByCurrentUser(inf.User.UserName, inf.Tx.Tx, &profile.CDNName, &profile.CDNID)
+		if userErr != nil || sysErr != nil {
+			api.HandleErr(w, r, tx, statusCode, userErr, sysErr)
+			return
+		}
+	}
+
+	requestedProfileId := inf.IntParams["id"]
+	// check if the entity was already updated
+	userErr, sysErr, errCode = api.CheckIfUnModified(r.Header, inf.Tx, requestedProfileId, "profile")
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+
+	//update profile
+	query := `UPDATE profile SET
+		name = $2, 
+		cdn = $3,
+		type = $4, 
+		routing_disabled = $5, 
+		description = $6
+	WHERE id = $1
+	RETURNING id, last_updated, name, description, (select name FROM cdn where id = $3), cdn, routing_disabled, type`
+
+	err := tx.QueryRow(query, requestedProfileId, profile.Name, profile.CDNID, profile.Type, profile.RoutingDisabled, profile.Description).
+		Scan(&profile.ID, &profile.LastUpdated, &profile.Name, &profile.Description, &profile.CDNName, &profile.CDNID, &profile.RoutingDisabled, &profile.Type)
+
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			api.HandleErr(w, r, tx, http.StatusBadRequest, fmt.Errorf("profile: %s not found", profile.Name), nil)
+			return
+		}
+		usrErr, sysErr, code := api.ParseDBError(err)
+		api.HandleErr(w, r, tx, code, usrErr, sysErr)
+		return
+	}
+
+	alerts := tc.CreateAlerts(tc.SuccessLevel, "profile was updated")
+	api.WriteAlertsObj(w, r, http.StatusOK, alerts, profile)
+	return
+}
+
+// Delete an profile for APIv5
+func Delete(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	tx := inf.Tx.Tx
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	id := inf.Params["id"]
```

Comment:
This should be a required parameter as passed into `api.NewInfo`. If that is done, then it can also be specified as an integral parameter, which will allow you to access it as `inf.IntParams["id"]`, meaning you can skip the `strconv.Atoi` step.